contractsV5/KyberHintHandler.sol

```diff
@@ -20,126 +25,82 @@ contract KyberHintParser is IKyberHint, Utils {
         TradeType tradeType;
         IKyberReserve[] addresses;
         uint[] splitValuesBps;
+        bool invalidHint;
     }
 
     struct TradeHint {
         ReservesHint ethToTokenReserves;
         ReservesHint tokenToEthReserves;
     }
 
-    function parseEthToTokenHint(bytes calldata hint)
-        external
-        view
-        returns(
-            TradeType ethToTokenType,
-            bytes8[] memory ethToTokenReserveIDs,
-            uint[] memory ethToTokenSplits,
-            uint failingIndex
-        )
-    {
-
-    }
-
-    function parseE2tHint(bytes memory hint)
+    function parseHintE2T(bytes memory hint)
         internal
         view
         returns(
-            TradeType ethToTokenType,
-            IKyberReserve[] memory ethToTokenAddresses,
-            uint[] memory ethToTokenSplits,
-            uint failingIndex
+            TradeType e2tType,
+            IKyberReserve[] memory e2tAddresses,
+            uint[] memory e2tSplits
         )
     {
         TradeHint memory tradeHint;
         uint indexToContinueFrom;
 
         decodeOperation(hint, tradeHint, indexToContinueFrom, false);
 
-        ethToTokenType = tradeHint.ethToTokenReserves.tradeType;
-        ethToTokenAddresses = tradeHint.ethToTokenReserves.addresses;        
-        ethToTokenSplits = tradeHint.ethToTokenReserves.splitValuesBps;
+        require(!tradeHint.ethToTokenReserves.invalidHint, "Invalid hint");
```

Comment:
seems to me if we can't return failing index.
at least we could revert with a clearer reason and some index. and not revert in one centralized place.
lets see how failing revert msgs can be as helpful as possible. also for our own testing.

Reply:
Haven't really worked on the failingIndex. Problem is revert/require reason must be a string and can't insert variables in the message.